share/change.py

```diff
@@ -193,7 +218,12 @@ def __init__(self, graph):
         self.__graph = graph
         self.__visted = set()
         self.__visiting = set()
-        self.__nodes = list(graph.nodes)
+
+        # visit identifiers first
+        self.__nodes = sorted(
```

Comment:
Relevant comment from above.

> This is a subtle but very important nuance. The order that these are returned in is critically import to the disambiguation process.

>There are a few other tests that check this but I'd sleep better know that there are many assertions for it.

> See test_topological_sort_many_to_many for another example.


Identifiers will always be first as they are referenced by creativeworks or by people and they,  themselves, don't reference anything. The purpose of the resolve relations function is to ensure that creativeworks have outgoing references to anything they are connected to by a through table.


```
            --------------------------------
           /                                \
CreativeWork <-- Through --> Identifier <----


```